src/renderer/components/chat/ChatList.tsx

```diff
@@ -401,34 +401,47 @@ export function useLogicVirtualChatList(chatListIds: [number, number][]) {
     })
   }
 
-  const onChatListItemChanged = useCallback(
-    (chatId: number, _messageId: number | string) => {
-      const debouncingChatlistItemRequests: { [chatid: number]: number } = {}
-      const updateChatListItem = async (chatId: number) => {
-        debouncingChatlistItemRequests[chatId] = 1
-        // the message id of the event could be an older message than the newest message (for example msg-read event)
-        const messageId = await DeltaBackend.call(
-          'chatList.getChatListEntryMessageIdForChatId',
-          chatId
-        )
-        setChatLoading(state => ({
-          ...state,
-          [chatId]: LoadStatus.FETCHING,
-        }))
-        const chats = await DeltaBackend.call(
-          'chatList.getChatListItemsByEntries',
-          [[chatId, messageId]]
-        )
-        setChatCache(cache => ({ ...cache, ...chats }))
-        setChatLoading(state => ({
-          ...state,
-          [chatId]: LoadStatus.LOADED,
-        }))
-        if (debouncingChatlistItemRequests[chatId] > 1) {
-          updateChatListItem(chatId)
-        } else {
-          debouncingChatlistItemRequests[chatId] = 0
-        }
+  const onChatListItemChanged = useMemo(() => {
```

Comment:
Why useMemo over useCallback?

Reply:
because then I can not define the stuff outside of the function, read the next lines (where `{` begin and end) then you'll understand